### PR TITLE
Rewrite Election views

### PR DESCRIPTION
**Pull request by davidwessman**
Updating Election, Candidates and Nominations controller.
Updating belonging views, making it easiser to navigate.
Removing obsolete methods.

### CODE REVIEW

File: app/models/election.rb [hunk 1]
```diff
@@ -24,49 +27,38 @@ def rest_grid
...
     else
       posts.title
     end
   end
 
+  def after_posts
+    case state
+    when :after
+      posts.title.general
+    end
```
Comment (jforberg) [inline app/models/election.rb]: state == :after ? posts.title.general : nil

File: app/services/election_service.rb [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
+      true
+    else
+      false
+    end
   end
 
   def self.destroy_candidate(candidate)
     if candidate.editable?
-      candidate.destroy!
+      candidate.destroy
```
Comment (jforberg) [inline app/services/election_service.rb]: What happens if this fails?

File: app/view_objects/election_view.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
+class ElectionView
+  attr_reader :election
+  attr_accessor :grid, :rest_grid, :candidates, :candidate, :user, :nomination
+
+  def initialize(election, candidate: nil, candidates: [])
+    @election = election
+    @candidate = candidate
+    @candidates = candidates
```
Comment (jforberg) [inline app/view_objects/election_view.rb]: Could candidate and candidates be better named to clarify their function?

**Comment by jforberg** [comment]
Good work. Some comments. -1

File: app/services/election_service.rb (hunk 2 above)
Comment (davidwessman) [inline app/services/election_service.rb]: https://github.com/fsek/web/blob/david-candidates/app/controllers/elections/candidates_controller.rb#L42

But I guess I could just as well do it with a bang.

**Comment by davidwessman** [comment]
@jforberg Please look through my additions 😄 

I added a CandidateView-object to make it clearer.
Changed to `destroy!`

**Comment by davidwessman** [comment]
If you have any other comments I'll fix them in smaller patches.
The election is opening today.

**Pull request merged**